### PR TITLE
Pay debts older than 20 days

### PR DESCRIPTION
**Pull request by g4s8**
#696 - pay debts if it is older than `46.days` days

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#883](https://github.com/zerocracy/farm/issues/883) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/921427772); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

File: src/main/resources/com/zerocracy/stk/pmo/debts/pay_all_debts.groovy [hunk 1]
```diff
@@ -42,7 +49,10 @@ def exec(Project pmo, XML xml) {
       return
     }
     Cash debt = debts.amount(uid)
-    if (debt < new Policy().get('46.threshold', new Cash.S('$50'))) {
+    Policy policy = new Policy()
+    int
```
Comment (carlosmiranda) [inline src/main/resources/com/zerocracy/stk/pmo/debts/pay_all_debts.groovy]: @g4s8 what's this? :)

File: src/main/resources/com/zerocracy/stk/pmo/debts/pay_all_debts.groovy [hunk 2]
```diff
@@ -42,7 +49,10 @@ def exec(Project pmo, XML xml) {
...
+    if (debt < policy.get('46.threshold', new Cash.S('$50')) &&
+      new Date().time - debts.oldest(uid).time < TimeUnit.DAYS.toMillis(policy.get('46.days', 20))) {
```
Comment (carlosmiranda) [inline src/main/resources/com/zerocracy/stk/pmo/debts/pay_all_debts.groovy]: @g4s8 a rather cumbersome boolean check. What if we add a method, something like `Debts.olderThan(uid, date)`, that we can call here instead?

**Comment by g4s8** [comment]
@carlosmiranda fixed, please see 354b1a6

**Comment by carlosmiranda** [comment]
@yegor256 ping

**Comment by 0crat** [comment]
@carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

File: src/main/resources/com/zerocracy/stk/pmo/debts/pay_all_debts.groovy [hunk 3]
```diff
@@ -42,7 +49,9 @@ def exec(Project pmo, XML xml) {
       return
     }
     Cash debt = debts.amount(uid)
-    if (debt < new Policy().get('46.threshold', new Cash.S('$50'))) {
+    Policy policy = new Policy()
+    if (debt < policy.get('46.threshold', new Cash.S('$50')) &&
+      debts.olderThan(uid, new Date(new Date().time - TimeUnit.DAYS.toMillis(policy.get('46.days', 20))))) {
```
Comment (yegor256) [inline src/main/resources/com/zerocracy/stk/pmo/debts/pay_all_debts.groovy]: @g4s8 I believe it should be "NOT olderThan"

**Comment by yegor256** [comment]
@g4s8 see above

**Comment by g4s8** [comment]
@yegor256 right, fixed

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#883](https://github.com/zerocracy/farm/issues/883) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**